File: 0357-Count Numbers with Unique Digits/0357-Count Numbers with Unique Digits.cpp
class Solution {
public:
    int countNumbersWithUniqueDigits(int n) {
        int count = 1;
        int product = 9;
        int base = 9;
        for (int i = 1; i <= n && i <= 10; ++i)
        {
            count += product;
            product *= base;
            --base;
        }
        
        return count;
    }
};

File: 0301-0400/0357-Count Numbers with Unique Digits/0357-Count Numbers with Unique Digits.py
class Solution:
    def countNumbersWithUniqueDigits(self, n: int) -> int:
        count = 1
        base = 9
        product = 9
        for i in range(1, min(n+1, 11)):
            count += product
            product *= base
            base -= 1
        return count
